lib/http.js

```diff
@@ -4,15 +4,26 @@ var cookies = require('./cookies');
 var Q = require('q');
 var AuthApiError = require('./errors/AuthApiError');
 var config = require('./config');
+var storageBuilder = require('./storageBuilder');
+
+var httpCache = storageBuilder(localStorage, config.CACHE_STORAGE_NAME);
 
 function httpRequest(sdk, options) {
   options = options || {};
   var url = options.url,
       method = options.method,
       args = options.args,
-      dontSaveResponse = options.dontSaveResponse,
+      cacheState = options.cacheState,
       accessToken = options.accessToken;
 
+  if (options.responseCacheDuration) {
+    var cacheContents = httpCache.getStorage();
+    var cachedResponse = cacheContents[url];
+    if (cachedResponse && Date.now()/1000 < cachedResponse.expiresAt) {
```

Comment:
I'm also curious how we're planning on handling the keys call, specifically:
1. First time user, gets keys, it's cached
2. Second time coming in, cached and kid is still there (no problem)
3. Keys rotate
4. Make request, is cached, kid is not there; make request again <-- what happens if for whatever reason the kid is not there?

I.e. it doesn't look like we expose whether we retrieved the response from the cache or not - is that going to be a problem when we need to have logic like "see what we have, looks like it's not there, make a request to get the new info".